app/models/miq_cockpit_ws_worker/runner.rb

```diff
@@ -0,0 +1,196 @@
+    if cockpit_ws_alive?
+      @stdout.readpartial(1.megabyte).split("\n").each { |msg| $log.info  "cockpit-ws: #{msg.rstrip}" } if @stdout && @stdout.ready?
+      @stderr.readpartial(1.megabyte).split("\n").each { |msg| $log.error "cockpit-ws: #{msg.rstrip}" } if @stderr && @stderr.ready?
+    else
+      _log.info("#{log_prefix} Cockpit-ws Process gone. Restarting...")
+      start_cockpit_ws
+    end
+  end
+
+  def start_cockpit_ws
+    raise _("Cannot call start_cockpit_ws if a process already exists") if cockpit_ws_alive?
+    _log.info("#{log_prefix} Starting cockpit-ws Process")
+    start_cockpit_ws_process
+    _log.info("#{log_prefix} Started cockpit-ws Process")
+  end
+
+  def start_cockpit_ws_process
+    @pid, @stdout, @stderr = cockpit_ws_run
+  rescue => err
+    _log.error("#{log_prefix} cockpit-ws Process aborted because [#{err.message}]")
+    _log.log_backtrace(err)
+  end
+
+  def stop_cockpit_ws
+    if cockpit_ws_alive?
+      _log.info("#{log_prefix} Shutting down cockpit-ws process...pid=#{@pid}")
+      stop_cockpit_ws_process
+      _log.info("#{log_prefix} Shutting down cockpit-ws process...Complete")
+    end
+  end
+
+  def stop_cockpit_ws_process
+    return unless @pid
+    Process.kill("INT", @pid)
+    wait_on_cockpit_ws
+  end
+
+  # Waits for a cockpit-ws process to stop.  The process is expected to be
+  #   in the act of shutting down, and thus it will wait 5 minutes
+  #   before issuing a kill.
+  def wait_on_cockpit_ws
+    # TODO: Use Process.waitpid or one of its async variants
+    begin
+      Timeout.timeout(5.minutes.to_i) do
+        loop do
+          break unless process_alive?(@pid)
+          sleep 1
+          heartbeat
+        end
+      end
+    rescue Timeout::Error
+      _log.info("#{log_prefix} Killing cockpit-ws process with pid=#{@pid}")
+      Process.kill(9, @pid)
+    end
+
+    _log.info("#{log_prefix} cockpit-ws Waiting for process with pid=#{@pid}")
+    pid, status = Process.waitpid2(@pid)
+    _log.info("#{log_prefix} cockpit-ws Process with pid=#{pid} exited with a status=#{status}")
+
+    reset_process_info
+  end
+
+  def reset_process_info
+    @pid = @stdout = @stderr = nil
+  end
+
+  def find_cockpit_ws_processes
+    find_pids(/cockpit-ws --port/)
+  end
+
+  def stop_active_cockpit_ws_processes
+    find_cockpit_ws_processes.each do |pid|
+      _log.info("#{log_prefix} Killing active cockpit-ws process with pid=#{pid}")
+      Process.kill(9, pid)
+    end
+  end
+
+  def cockpit_ws_alive?
+    if process_alive?(@pid)
+      true
+    else
+      reset_process_info
+      false
+    end
+  end
+
+  def process_alive?(pid)
+    begin
+      stat = MiqProcess.linux_process_stat(pid) unless pid.nil?
+    rescue SystemCallError => err
+      _log.error("#{log_prefix} cockpit-ws Process with pid=#{pid} '#{err.message}' while checking process state")
+      return false
+    end
+
+    return false if stat.nil?
+    pid_state = LINUX_STATES[stat[:state].to_s.first]
```

Comment:
against, try to see if we can use existing code in Sys::ProcTable... there's a `state` method on the struct returned from `ps`.

https://github.com/djberg96/sys-proctable/blob/2bc2fb9b42e8c1c8effd53ced186e1afb7c33428/lib/linux/sys/proctable.rb#L31